tombstones.go

```diff
@@ -156,30 +157,55 @@ func readTombstones(dir string) (memTombstones, error) {
 	return stonesMap, nil
 }
 
-type memTombstones map[uint64]Intervals
+type memTombstones struct {
+	mts map[uint64]Intervals
+	mtx sync.RWMutex
+}
+
+func newEmptyMemTombstones() *memTombstones {
+	mts := make(map[uint64]Intervals)
+	return &memTombstones{mts: mts}
+}
 
-var emptyTombstoneReader = memTombstones{}
+func newMemTombstones(mts map[uint64]Intervals) *memTombstones {
+	return &memTombstones{mts: mts}
+}
+
+//var emptyTombstoneReader = memTombstones{}
+var emptyTombstoneReader = newEmptyMemTombstones()
 
 // EmptyTombstoneReader returns a TombstoneReader that is always empty.
 func EmptyTombstoneReader() TombstoneReader {
 	return emptyTombstoneReader
 }
 
-func (t memTombstones) Get(ref uint64) (Intervals, error) {
-	return t[ref], nil
+func (t *memTombstones) Get(ref uint64) (Intervals, error) {
+	t.mtx.RLock()
+	defer t.mtx.RUnlock()
+	return t.mts[ref], nil
 }
 
-func (t memTombstones) Iter(f func(uint64, Intervals) error) error {
-	for ref, ivs := range t {
+func (t *memTombstones) Iter(f func(uint64, Intervals) error) error {
+	t.mtx.Lock()
+	defer t.mtx.Unlock()
+	for ref, ivs := range t.mts {
 		if err := f(ref, ivs); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func (t memTombstones) add(ref uint64, itv Interval) {
-	t[ref] = t[ref].add(itv)
+func (t *memTombstones) add(ref uint64, itv Interval) {
+	t.mtx.Lock()
+	t.mts[ref] = t.mts[ref].add(itv)
+	t.mtx.Unlock()
+}
+
+func (t *memTombstones) put(ref uint64, itvs Intervals) {
+	t.mtx.Lock()
+	t.mts[ref] = itvs
+	t.mtx.Unlock()
 }
```

Comment:
hmm.. `add` and `put` are a bit confusing , need to think of some better naming.

Reply:
Maybe `add` ->`addInterval` and `put` ->`addIntervals`?

Reply:
hm ,seems we can use a single  Variadic Function.
`func (t *memTombstones) add(ref uint64, itv ...Interval)`

https://gobyexample.com/variadic-functions

Reply:
`add()` append an `Interval` to the an exsiting `Intervals`, while `put()` replace the old `Intervals`. It's a not good idea to combine them.

Reply:
I can have a look in few days, but there must be a better way for this.